### PR TITLE
Machine bash completion

### LINKED ISSUE #3620: Github URL for bash completion is wrong
File: [machine/completion.md](https://docs.docker.com/machine/completion/), CC @londoncalling 

The url of bash completion scripts in instructions (box with curl) is pointing to a wrong location.

I've did a search in repo and appear that the correct url is `https://raw.githubusercontent.com/docker/docker-ce/master/components/engine/contrib/completion/bash/docker` for bash, and `https://raw.githubusercontent.com/docker/docker-ce/master/components/cli/contrib/completion/zsh/_docker` for zsh

Thanks.

### LINKED ISSUE #3621: Wrong URL in compose's bash completion
File: [compose/completion.md](https://docs.docker.com/compose/completion/), CC @londoncalling 

Same problem of #3620. Suggested URLs for correction:

- `https://raw.githubusercontent.com/docker/compose/master/contrib/completion/bash/docker-compose` for bash
- `https://raw.githubusercontent.com/docker/compose/master/contrib/completion/zsh/_docker-compose` for zsh

Thanks

### PR DESCRIPTION
**Pull request by londoncalling**
#### Machine topics

- replaced incorrect bash and zsh completion URLs with correct ones per user feedback
- added list of available options once completion is installed, and a "What's next" section
- added links to Docker Cloud docs from Machine cloud topics in a note ("Try out Docker Cloud")

#### Compose topics

- replaced incorrect bash and zsh completion URLs with correct ones per user feedback
- added links to Docker Cloud docs from Machine cloud topics in a note ("Try out Docker Cloud")

#### Cloud and Mac/Windows topics

- Improved Cloud / Mac/ Windows cross referencing.

### Related

Fixes #3620 for Machine
Fixes #3621 for Compose

### Reviewers

@trestini @shin- @ManoMarks @mstanleyjones 

Signed-off-by: Victoria Bialas <victoria.bialas@docker.com>

### CODE REVIEW

File: compose/completion.md [hunk 1]
```diff
@@ -11,54 +11,72 @@ for the bash and zsh shell.
...
+non-minimal installation, bash completion should be available. On a Mac, install
+with `brew install bash-completion`.
 
-Place the completion script in `/etc/bash_completion.d/` (`/usr/local/etc/bash_completion.d/` on a Mac), using e.g.
+Place the completion script in `/etc/bash_completion.d/`
+(`/usr/local/etc/bash_completion.d/` on a Mac):
 
-    curl -L https://raw.githubusercontent.com/docker/compose/$(docker-compose version --short)/contrib/completion/bash/docker-compose -o /etc/bash_completion.d/docker-compose
+```shell
+curl -L https://raw.githubusercontent.com/docker/compose/master/contrib/completion/bash/docker-compose -o /etc/bash_completion.d/docker-compose
```
Comment (mstanleyjones) [inline compose/completion.md]: You will need sudo on this. Is it possible to put it somewhere local to the user, like the ZSH example below, instead?

Comment (londoncalling) [inline compose/completion.md]: I'll try it.

Comment (londoncalling) [inline compose/completion.md]: it already says to do that

**Comment by londoncalling** [comment]
LGTM

**Pull request merged**